### PR TITLE
Fix the version lock update code

### PR DESCRIPTION
**Pull request by rw-access**
## Issues
Unable to save the version lock

## Summary
I noticed that the most recent changes in #1029 were made after I tested the lock.
I think I never tried re-saving the lock in the new branch. Some of the code was moved.

Now you can save the lock again.
Package hash still looks good.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: detection_rules/packaging.py [hunk 1]
```diff
@@ -112,7 +112,18 @@ def manage_versions(rules: List[TOMLRule], deprecated_rules: list = None, curren
 
         if save_changes:
             if changed_rules or (new_rules and add_new):
-                current_versions.update(new_rules if add_new else {})
+                if add_new:
+                    for rule in new_rules:
+                        current_versions[rule.id] = {"rule_name": rule.name,
```
Comment (brokensound77) [inline detection_rules/packaging.py:117]: nit: the dict contents should start on a new, indented line

**Review by brokensound77 — APPROVED** [review]
Glad you found this quickly!

File: detection_rules/packaging.py (hunk 1 above)
Comment (rw-access) [reply detection_rules/packaging.py:117]: alternate idea: build the dict earlier where it's not indentend so far and save 5 lines

Comment (rw-access) [reply detection_rules/packaging.py]: added a method to return the dict e60fab52bb263433b9a9f0fb4edc3c7766b4626c

**Pull request merged**